modules/tfm/tfm/boards/src/tfm_ioctl_s_api.c

```diff
@@ -34,3 +34,31 @@ enum tfm_platform_err_t tfm_platform_mem_read(void *destination, uint32_t addr,
 
 	return (enum tfm_platform_err_t) ret;
 }
+
+__attribute__((section("SFN")))
+enum tfm_platform_err_t tfm_platform_gpio_pin_mcu_select(uint32_t pin_number, uint32_t mcu,
+							 uint32_t *result)
+{
+	enum tfm_platform_err_t ret;
+	psa_invec in_vec;
+	psa_outvec out_vec;
+	struct tfm_gpio_service_args args;
+	struct tfm_gpio_service_out out;
+
+	args.type = TFM_GPIO_SERVICE_TYPE_PIN_MCU_SELECT;
+	args.mcu_select.pin_number = pin_number;
+	args.mcu_select.mcu = mcu;
+
+	in_vec.base = (const void *)&args;
+	in_vec.len = sizeof(args);
+
+	out_vec.base = (void *)&out;
+	out_vec.len = sizeof(out);
+
+	ret = tfm_platform_ioctl(TFM_PLATFORM_IOCTL_GPIO_SERVICE, &in_vec,
+				 &out_vec);
+
+	*result = out.result;
+
+	return ret;
+}
```

Comment:
Let's use ifdef's to enforce zero flash usage for platforms that don't have MCUSEL (nrf91).

Reply:
Do you have any opinion on what is the proper return value for platforms that don't have MCUSEL?
At the moment it is
`TFM_PLATFORM_ERR_SUCESS, err = -1`, but maybe this should be `TFM_PLATFORM_ERR_NOT_SUPPORTED`?

Reply:
No opinion.

Reply:
Returns not supported error code now.
Also updated the GPIO service to check the type argument, which I had forgotten.